src/index.ts

```diff
@@ -0,0 +1,43 @@
+import fetch from 'cross-fetch';
+import startsWith from 'lodash.startswith';
+
+type FetchClient = (
+  input: RequestInfo,
+  init?: RequestInit | undefined,
+) => Promise<Response>;
+
+async function checkResponseIsOk(response: Response): Promise<Response> {
+  if (response.ok) {
+    return response;
+  }
+  const error = await response.text();
```

Comment:
AFAIK, this is the same as `return response.text()`. And therefore you can make this function be only:

```js
const checkResponseIsOk = (response: Response): Promise<Response> => response.ok ? response : response.text();
```

Reply:
Maybe a good comment would be welcome here.